src/librustc_middle/mir/interpret/allocation.rs

```diff
@@ -368,7 +364,9 @@ impl<'tcx, Tag: Copy, Extra: AllocationExtra<Tag>> Allocation<Tag, Extra> {
             return Ok(ScalarMaybeUninit::Uninit);
         }
         // Now we do the actual reading.
-        let bits = read_target_uint(cx.data_layout().endian, bytes).unwrap();
+        let mut buf = [0; 16];
+        bytes.read(&mut buf).unwrap();
+        let bits = u128::from_ne_bytes(buf);
```

Comment:
is this correct? before we read as `cx.data_layout().endian` order.

Reply:
I backed that commit out after discovering that I had somehow messed up testing it anyways because I didn't notice other parts of librustc were depending on it. ^_^;;

It actually should be correct though, given that {from, to}_ne_bytes is supposed to be target endianness and the target endianness is what that references.

Reply:
Not quite, `ne` is native endian of the host that target compiler was built. `cx.data_layout().endian` is the target endian that compiler produces code for.

Reply:
Right, I am saying I am pretty sure that it's the other way around `ne` is target, not host, given it says target [right here](https://doc.rust-lang.org/nightly/std/primitive.u128.html#method.from_ne_bytes).

Reply:
Note that _build_, _host_ and _target_ are confusing terms in [GNU build system](https://gcc.gnu.org/onlinedocs/gccint/Configure-Terms.html) and different with the meaning in Rust build.
The **ne** method say 
> As the target platform's native endianness

"target" here is the host that the code uses _ne_ was built, in this case the compiler.